### PR TITLE
Data 399 SurveyMonkey tap v2.0.0

### PR DESCRIPTION
**Pull request by graham-jackson**
Work in progress on updating the Netsuite tap

- Added pytest and integration tests for discovery and sync
- Bumped major version since tap will 'discover' more/different streams
- Added incomplete `sync_surveys` functionality WIP; will create follow up ticket
- Changed default discover behaviour to be that tables are for `FULL_TABLE` replication
- Added `schemas/surveys.json`

Cross checks
- `make test` works
- `make discover` works
- `make upload` works

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,7 @@
 .env
 .sentinel*
 .venv/*
+/catalog-discovered.json
```
Comment (adrien-mg) [inline .gitignore:4]: Out of curiosity, does the `/` mean anything here?

**Comment by adrien-mg** [comment]
The title is misleading, this is not Netsuite it's SurveyMonkey!

**Review by adrien-mg — APPROVED** [review]
Yep, looks good

**Comment by adrien-mg** [comment]
Just added a small commit to remove unused requirements.txt, it's not used anywhere and all dependencies are in the `setup.py` file.

File: .gitignore (hunk 1 above)
Comment (graham-jackson) [reply .gitignore:4]: It's telling git to ignore files called `catalog-discovered.json` in the root folder only.

cf.
```
graham.jackson:tmp$ cat .gitignore
.gitignore
foo.txt
graham.jackson:tmp$ tree
.
├── foo.txt
└── my_package
    └── tests
        └── foo.txt

2 directories, 2 files
graham.jackson:tmp$ git status
On branch main

No commits yet

nothing to commit (create/copy files and use "git add" to track)
graham.jackson:tmp$ cat .gitignore
.gitignore
/foo.txt
graham.jackson:tmp$ git status
On branch main

No commits yet

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	my_package/

nothing added to commit but untracked files present (use "git add" to track)
```

Comment (graham-jackson) [reply .gitignore:4]: ...I.e. initially the gitignore contains a line `foo.txt` so both `foo.txt` files are ignored. 

Then I update the gitignore to contain `/foo.txt` and so the root one is ignored, but the one in `my_package/tests` isn't.

**Comment by graham-jackson** [comment]
> The title is misleading, this is not Netsuite it's SurveyMonkey!

🤦 
I'll see if I can change it.

**Review by graham-jackson — COMMENTED** [review]
LGTM

**Pull request merged**